opencodeblocks/graphics/view.py

```diff
@@ -131,13 +141,44 @@ def leftMouseButtonRelease(self, event: QMouseEvent):
     def rightMouseButtonPress(self, event: QMouseEvent):
         """ OCBView reaction to rightMouseButtonPress event. """
         event = self.drag_scene(event, "press")
+        self.right_click_time = time.time()
         super().mousePressEvent(event)
 
     def rightMouseButtonRelease(self, event: QMouseEvent):
         """ OCBView reaction to rightMouseButtonRelease event. """
         event = self.drag_scene(event, "release")
         super().mouseReleaseEvent(event)
         self.setDragMode(QGraphicsView.DragMode.RubberBandDrag)
+        if time.time() - self.right_click_time < RIGHT_CLICK_SPEED:
+            self.fastContextMenuEvent(event)
+
+    def fastContextMenuEvent(self, event: QContextMenuEvent):
+        """
+        Displays the context menu when inside a view
+
+        We don't use the default contextMenuEvent method to avoid
+        displaying a context menu when moving around the view.
+
+        For this method to be triggered, the click has to last less than RIGHT_CLICK_SPEED.
+        """
+        menu = QMenu(self)
+        actionPool = []
+        blockTypes = os.listdir("blocks")
+        for b in blockTypes:
+            filepath = os.path.join("blocks", b)
+            with open(filepath, "r", encoding="utf-8") as file:
+                data = json.loads(file.read())
+                if "title" not in data:
+                    actionPool.append((filepath, menu.addAction(f"New Block")))
+                else:
+                    actionPool.append(
+                        (filepath, menu.addAction(f"New {data['title']} Block")))
+
+        selectedAction = menu.exec_(self.mapToGlobal(event.pos()))
+        for filepath, action in actionPool:
+            if action == selectedAction:
+                p = self.mapToScene(event.pos())
+                self.scene().create_block_from_file(filepath, p.x(), p.y())
```

Comment:
To avoid this conflict, we might want to switch the scene dragging around to the mouse middle click instead of right click.

Reply:
Mostly because we can drag the scene around with two fingers on laptop anyway, so we don't need the right clic to be used for that anymore